diesel_tests/tests/only.rs

```diff
@@ -0,0 +1,236 @@
+}
+
+fn test_scenario(connection: &mut PgConnection) {
+    // Two users - one is archived.
+    // Each user has two posts - one archived each.
+    let uid: i32 = diesel::insert_into(users::table)
+        .values(users::name.eq("Sean"))
+        .returning(users::id)
+        .get_result(connection)
+        .unwrap();
+    assert_eq!(uid, 1);
+
+    let uid: i32 = diesel::insert_into(users_archived::table)
+        .values((
+            users_archived::name.eq("Tess"),
+            users_archived::archived.eq(true),
+        ))
+        .returning(users_archived::id)
+        .get_result(connection)
+        .unwrap();
+    assert_eq!(uid, 2);
+
+    for user_id in [1, 2] {
+        diesel::insert_into(posts::table)
+            .values((posts::user_id.eq(user_id), posts::title.eq("Post")))
+            .execute(connection)
+            .unwrap();
+        diesel::insert_into(posts_archived::table)
+            .values((
+                posts_archived::user_id.eq(user_id),
+                posts_archived::title.eq("Archived post"),
+                posts_archived::archived.eq(true),
+            ))
+            .execute(connection)
+            .unwrap();
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Queryable, Clone, Insertable, AsChangeset, Selectable)]
+#[diesel(table_name = users)]
+pub struct NewUser {
+    pub name: String,
+}
+
+#[test]
+fn select_from_only_with_inherited_table() {
+    let connection = &mut connection();
+    setup_tables(connection);
+    test_scenario(connection);
+
+    // There is now only one entry in the users_archived table, none in the users table.
+
+    let n_users = users::table
+        .select(count(users::id))
+        .first::<i64>(connection)
+        .unwrap();
+    assert_eq!(n_users, 2);
+
+    let n_users_in_main_table = users::table
+        .only()
+        .select(count(users::id))
+        .first::<i64>(connection)
+        .unwrap();
+    assert_eq!(n_users_in_main_table, 1);
+}
+
+#[test]
+fn select_from_only_filtering_and_find() {
+    // Test that it's possible to call `.only().filter(..)`
+    let connection = &mut connection();
+    setup_tables(connection);
+    test_scenario(connection);
+
+    assert_eq!(
+        users::table
+            .only()
+            .filter(users::name.eq("Sean"))
+            .select(users::name)
+            .load::<String>(connection)
+            .unwrap()
+            .len(),
+        1
+    );
+    assert_eq!(
+        users::table
+            .only()
+            .filter(users::name.eq("Tess"))
+            .select(users::name)
+            .load::<String>(connection)
+            .unwrap()
+            .len(),
+        0
+    );
+    assert_eq!(
+        users::table
+            .only()
+            .find(1)
+            .count()
+            .get_result::<i64>(connection)
+            .unwrap(),
+        1
+    );
+}
+
+#[test]
+fn inner_join_only() {
+    // Test that it's possible to call `.inner_join(X::table.only().on(...)`
```

Comment:
Specifically this test is missing. I suggest that we either remove that comment or add a corresponding test :wink: